apps/wallet-mobile/src/features/Receive/common/AddressDetailCard/AddressDetailCard.tsx

```diff
@@ -21,52 +21,47 @@ type AddressDetailsProps = {
 }
 
 type CardItem = {
-  title: ShareProps['title']
-  address: ShareProps['address']
+  title: AddressDetailCardProps['title']
+  address: string
 } & (
   | {
       cardType: 'QRCode'
     }
   | ({cardType: 'Details'} & AddressDetailsProps)
 )
 
-export const AddressDetailCard = ({title}: ShareProps) => {
-  const [scrollPosition, setScrollPosition] = useState(0)
-  const [isSecondPage, setIsSecondPage] = useState(false)
-
-  const {selectedAddress} = useReceive()
-
-  const [isCopying, copy] = useCopy()
-
+export const AddressDetailCard = ({title}: AddressDetailCardProps) => {
   const {styles, colors} = useStyles()
+  const [isCopying, copy] = useCopy()
 
-  const SCREEN_WIDTH = useWindowDimensions().width
-  const itemsPerPage = 1
-
-  const keyHashes = useKeyHashes({address: selectedAddress ?? ''})
+  const {selectedAddress: address} = useReceive()
+  const {spending, staking} = useKeyHashes({address})
+  const stakingHash = staking ?? ''
+  const spendingHash = spending ?? ''
 
+  const [scrollPosition, setScrollPosition] = React.useState(0)
   const cards: ReadonlyArray<CardItem> = [
-    {cardType: 'QRCode', title, address: selectedAddress ?? ''},
+    {cardType: 'QRCode', title, address},
     {
       cardType: 'Details',
-      address: selectedAddress ?? '',
-      stakingHash: keyHashes?.staking ?? '',
-      spendingHash: keyHashes?.spending ?? '',
+      address,
+      stakingHash,
+      spendingHash,
       title,
     },
   ] as const
-
+  const screenWidth = useWindowDimensions().width
+  const itemsPerPage = 1
+  const minToSwitchPage = 64
   const totalPages = Math.ceil(cards.length / itemsPerPage)
-  const circleIndex = Array.from({length: totalPages}, (_, index) => index + 1)
+  const cardIndicators = Array.from({length: totalPages}, (_, index) => index)
+
+  if (isEmptyString(address)) return
 
   const handleOnPageChange = (event: {nativeEvent: {contentOffset: {x: number}}}) => {
     const offset = event.nativeEvent.contentOffset.x
-    const index = Math.floor(offset / (itemsPerPage * SCREEN_WIDTH - 64))
-    setScrollPosition(index)
-  }
-
-  if (selectedAddress === undefined) {
-    return
+    const index = Math.floor(offset / (itemsPerPage * screenWidth - minToSwitchPage))
+    setScrollPosition(Math.max(index, 0))
```

Comment:
otherwise when swiping left at first page will cause no index be off while holding the first page, causing the indicator to be wrong until the page is released.